src/Tasks/Microsoft.NET.Build.Tasks/RunReadyToRunCompiler.cs

```diff
@@ -149,16 +200,36 @@ private string GetAssemblyReferencesCommands()
             return result.ToString();
         }
 
+        protected override string GenerateCommandLineCommands()
+        {
+            if (UseCrossgen2 && !string.IsNullOrEmpty(DotNetHostPath))
+            {
+                return Quote(Crossgen2Tool.ItemSpec);
+            }
+            return null;
+        }
+
+        private static string Quote(string path)
+        {
+            if (string.IsNullOrEmpty(path) || (path[0] == '\"' && path[path.Length - 1] == '\"'))
```

Comment:
I do not think we should introduce this function.  We have already checked that `File.Exists(Crossgen2Tool.ItemSpec) == true`, which means `Crossgen2Tool.ItemSpec` cannot be `null`, an empty string, or quoted. Furthermore, quoting arbitrary paths (including directory paths) would require special handling of backslashes before quotes, which we'd better avoid.

Reply:
Removed in 6th commit.